pallets/automation-time/src/lib.rs

```diff
@@ -801,30 +834,32 @@ pub mod pallet {
 			Self::deposit_event(Event::TaskCancelled { who: task.owner_id, task_id });
 		}
 
-		/// Schedule task and return it's task_id.
+		/// Schedule tasks and return it's task_id.
 		pub fn schedule_task(
 			owner_id: AccountOf<T>,
 			provided_id: Vec<u8>,
-			time: u64,
+			time: Vec<UnixTime>,
 		) -> Result<T::Hash, Error<T>> {
 			let task_id = Self::generate_task_id(owner_id.clone(), provided_id.clone());
 
 			if let Some(_) = Self::get_task(task_id) {
 				Err(Error::<T>::DuplicateTask)?
 			}
 
-			match Self::get_scheduled_tasks(time) {
-				None => {
-					let task_ids: BoundedVec<T::Hash, T::MaxTasksPerSlot> =
-						vec![task_id].try_into().unwrap();
-					<ScheduledTasks<T>>::insert(time, task_ids);
-				},
-				Some(mut task_ids) => {
-					if let Err(_) = task_ids.try_push(task_id) {
-						Err(Error::<T>::TimeSlotFull)?
-					}
-					<ScheduledTasks<T>>::insert(time, task_ids);
-				},
+			for scheduled_time in time.iter() {
+				match Self::get_scheduled_tasks(*scheduled_time) {
```

Comment:
i forget if deferenced pointer clones the value for ya before it mutates it. prob good to unit test, but otherwise, this works for me!